internal/usecases/liquidity_provider/get_trusted_account.go

```diff
@@ -10,28 +11,45 @@ import (
 type GetTrustedAccountUseCase struct {
 	trustedAccountRepository liquidity_provider.TrustedAccountRepository
 	hashFunction             entities.HashFunction
+	signer                   entities.Signer
 }
 
 func NewGetTrustedAccountUseCase(
 	trustedAccountRepository liquidity_provider.TrustedAccountRepository,
 	hashFunction entities.HashFunction,
+	signer entities.Signer,
 ) *GetTrustedAccountUseCase {
 	return &GetTrustedAccountUseCase{
 		trustedAccountRepository: trustedAccountRepository,
 		hashFunction:             hashFunction,
+		signer:                   signer,
 	}
 }
 
 func (useCase *GetTrustedAccountUseCase) Run(ctx context.Context, address string) (*entities.Signed[liquidity_provider.TrustedAccountDetails], error) {
-	signedAccount, err := useCase.trustedAccountRepository.GetTrustedAccount(ctx, address)
-	if err != nil {
-		return nil, err
-	}
-	if signedAccount == nil {
-		return nil, liquidity_provider.ErrTrustedAccountNotFound
+	readFunction := func() (*entities.Signed[liquidity_provider.TrustedAccountDetails], error) {
+		signedAccount, err := useCase.trustedAccountRepository.GetTrustedAccount(ctx, address)
+		if err != nil {
+			return nil, err
+		}
+		if signedAccount == nil {
+			return nil, liquidity_provider.TrustedAccountNotFoundError
+		}
+		return signedAccount, nil
 	}
-	if err := signedAccount.CheckIntegrity(useCase.hashFunction); err != nil {
-		return nil, liquidity_provider.ErrTamperedTrustedAccount
+	signedAccount, err := liquidity_provider.ValidateConfiguration(
+		useCase.signer,
+		readFunction,
+		useCase.hashFunction,
+	)
+	if err != nil {
```

Comment:
errors.Is returns false if the err is nil, there is no need of checking first if error != nil, you can skip that to avoid the nesting and have something like:
```
if errors.Is(err, liquidity_provider.TrustedAccountNotFoundError) {
    return nil, err
} else if errors.Is(err, liquidity_provider.ConfigurationNotFoundError) {
    return nil, liquidity_provider.TrustedAccountNotFoundError
} else if err != nil {
    return nil, liquidity_provider.TamperedTrustedAccountError
}
```

<img width="459" alt="image" src="https://github.com/user-attachments/assets/2f72c44b-ec62-4a38-b9ce-db14937d55f4" />